modules/gapi/src/backends/ov/govbackend.cpp

```diff
@@ -1471,7 +1499,7 @@ void cv::gimpl::ov::GOVExecutable::run(cv::gimpl::GIslandExecutable::IInput  &in
     const auto &op = m_gm.metadata(this_nh).get<Op>();
 
     auto ctx = std::make_shared<OVCallContext>(uu, out, op.args, op.outs,
-            std::move(stub_meta), std::move(input_objs), std::move(output_objs));
+            std::move(stub_meta), std::move(input_objs), std::move(output_objs), m_inference_only);
```

Comment:
..and pass it as a whole thing here.

This method _was_ quite generic, but now it knows explicitly about `inference_only` mode - purely an abstraction leak.

Reply:
This method still doesn't know about `inference only` implementation details it just propagates "some" configuration into the kernel. But your point is correct, need to decide who exactly should be responsible for handling this mode:

There are a few options:

1. `Kernel` - This is the current approach. `inference_only` is propagated as the following:
    `compileArgs` -> `GOVExecutable` -> `OVCallContext` -> `Kernel` (e.g `Infer`, `InferList`, etc)

The benefit is that `IInferExecutor` that used as proxy to submit inference tasks don't know about this option at all, the `kernel` hides this under `set_input_data` and `read_output_data` callbacks so it may continue work in `sync`/`async` modes without knowledges about this mode.

2. `RequestPool / IInferExecutor` - In this apporach `GOVExecutable` may configure `RequestPool` the way it knows about benchmark mode.

Prons:
- Kernels don't know about this mode so they continue submit their tasks through `RequestPool` API.

Const:
- `inference_only` mode will be enabled for all `Kernel`'s (now it's only relevant for `Infer` (`InferList`, `InferROI`, `InferList2` throw exception if mode is enabled).
- Mode should be handled for both `SyncInferExecutor` and `AsyncInferExecutor`. This is a little bit tricky since currently `read_output_data` is also responsible for **posting** outputs to maintain contract with streaming executor.
https://github.com/TolyaTalamanov/opencv/blob/at/implement-inference-only-mode-for-ov-backend/modules/gapi/src/backends/ov/govbackend.cpp#L471

This approached looked much invasive than the first one.

Reply:
Yes and my point is to make `some configuration` really a `some configuration` but not a specific field.

Reply:
Done